heron/ckptmgr/src/java/com/twitter/heron/ckptmgr/CheckpointManagerServer.java

```diff
@@ -0,0 +1,301 @@
+    this.checkpointsBackend = checkpointsBackend;
+
+    this.connection = null;
+
+    registerInitialization();
+  }
+
+  private void registerInitialization() {
+    registerOnRequest(CheckpointManager.RegisterStMgrRequest.newBuilder());
+
+    registerOnRequest(CheckpointManager.RegisterTMasterRequest.newBuilder());
+
+    registerOnRequest(CheckpointManager.SaveInstanceStateRequest.newBuilder());
+
+    registerOnRequest(CheckpointManager.GetInstanceStateRequest.newBuilder());
+
+    registerOnRequest(CheckpointManager.CleanStatefulCheckpointRequest.newBuilder());
+  }
+
+  @Override
+  public void onConnect(SocketChannel channel) {
+    LOG.info("Got a new connection from host:port "
+        + channel.socket().getRemoteSocketAddress());
+  }
+
+  @Override
+  public void onRequest(REQID rid, SocketChannel channel, Message request) {
+    if (request instanceof CheckpointManager.RegisterStMgrRequest) {
+      handleStMgrRegisterRequest(rid, channel, (CheckpointManager.RegisterStMgrRequest) request);
+    } else if (request instanceof CheckpointManager.RegisterTMasterRequest) {
+      handleTMasterRegisterRequest(rid, channel,
+                                   (CheckpointManager.RegisterTMasterRequest) request);
+    } else if (request instanceof CheckpointManager.SaveInstanceStateRequest) {
+      handleSaveInstanceStateRequest(
+          rid, channel, (CheckpointManager.SaveInstanceStateRequest) request);
+    } else if (request instanceof CheckpointManager.GetInstanceStateRequest) {
+      handleGetInstanceStateRequest(
+          rid, channel, (CheckpointManager.GetInstanceStateRequest) request);
+    } else if (request instanceof CheckpointManager.CleanStatefulCheckpointRequest) {
+      handleCleanStatefulCheckpointRequest(
+          rid, channel, (CheckpointManager.CleanStatefulCheckpointRequest) request);
+    } else {
+      LOG.severe("Unknown kind of request: " + request.getClass().getName());
+    }
+  }
+
+  protected void handleCleanStatefulCheckpointRequest(
+      REQID rid,
+      SocketChannel channel,
+      CheckpointManager.CleanStatefulCheckpointRequest request
+  ) {
+    LOG.info(String.format("Got a clean request from %s running at host:port %s",
+             request.toString(), channel.socket().getRemoteSocketAddress()));
+
+    boolean deleteAll = request.hasCleanAllCheckpoints() && request.getCleanAllCheckpoints();
+    Common.StatusCode statusCode = Common.StatusCode.OK;
+
+    try {
+      checkpointsBackend.dispose(topologyName,
+                                 request.getOldestCheckpointPreserved(), deleteAll);
+      LOG.info("Dispose checkpoint successful");
+    } catch (StatefulStorageException e) {
+      LOG.log(Level.WARNING,
+              String.format("Request to dispose checkpoint failed for oldest Checkpoint "
+                            + "%s and deleteAll? %b",
+                            request.getOldestCheckpointPreserved(), deleteAll), e);
+      statusCode = Common.StatusCode.NOTOK;
```

Comment:
Whenever we set a status code that is not OK, can we also set Common.Status.message as well to give a brief description of what went wrong. This will help when troubleshooting a failed RPC from the client logs, by removing the need to have to inspect the RPC server logs and remove the need to try to correlate server logs with client logs.